File: imperium-common/src/main/java/com/xpdustry/imperium/common/account/MindustrySession.java
package com.xpdustry.imperium.common.account;

import java.net.InetAddress;
import java.time.Instant;

public record MindustrySession(Key key, int account, Instant createdAt, Instant expiresAt, Instant lastLogin) {
    public record Key(long uuid, long usid, InetAddress address) {}
}

File: imperium-common/src/main/kotlin/com/xpdustry/imperium/common/account/MindustrySession.kt
// SPDX-License-Identifier: GPL-3.0-only
package com.xpdustry.imperium.common.account

import java.time.Instant

data class MindustrySession(
    val key: SessionKey,
    val account: Int,
    val creation: Instant,
    val expiration: Instant,
    val lastLogin: Instant,
)

suspend fun MindustrySessionService.selectAccount(accounts: AccountService, key: SessionKey): Account? =
    selectByKey(key)?.let { accounts.selectById(it.account) }
